Pipeline/CreateUCDComponentVersion/dbb-ucd-packaging.groovy

```diff
@@ -134,31 +145,65 @@ properties.buildReportOrder.each{ buildReportFile ->
 		if(executeRecord.getOutputs().isEmpty() != true) {
 			count += executeRecord.getOutputs().size()
 			executeRecord.getOutputs().each{ output ->
-				def (ds,member) = getDatasetName(output.dataset)
+				def (ds,member) = getContainerAndFile(output.dataset)
 				tempBuildOutputsMap.put(new DeployableArtifact(member, output.deployType) , [ds, buildReport, executeRecord, buildReportRank])
 			}
 		}
 	}
 
+    // store build output information in Hashmap buildOutputsMap to replace potential duplicates
+    // managing TYPE USS_RECORD
+    USSRecords.each{ USSRecord ->
+        ArrayList<ArrayList> outputs = []
+        USSRecord.getAttribute("outputs").split(';').collectEntries { entry ->
+            outputs += entry.replaceAll('\\[|\\]', '').split(',')
```

Comment:
Can we use the `.add()` method to add to the output list. Difficult to read this line.

Reply:
I've tried, I couldn't find the right syntax with `add()`. It's using the groovy style of coding, not the Java style.